sql/rpl_parallel.h

```diff
@@ -362,6 +362,7 @@ struct rpl_parallel {
   HASH domain_hash;
   rpl_parallel_entry *current;
   bool sql_thread_stopping;
+  Relay_log_info *rli;
```

Comment:
Don't add this member just to have rli accessible from workers_idle(). Instead just pass the rli as an argument.
As an alternative, moving the workers_idle() into the Relay_log_info class may make more sense now, as you like.

Reply:
Yeah I went back and forth between these thoughts, and another alternative as a standalone/static function. I see pros/cons to all approaches, as having `rpl_parallel` be a direct member of the `Relay_log_info` feels awkward to then have to pass the `rli` _back_ to `rli.parallel.workers_idle(rli)`. I think moving the check into `Relay_log_info` makes the next most sense to me (though it feels to me as the concern of the `rpl_parallel` struct).

Though now another thought comes to mind as a static function of `rpl_parallel`, which I think I am happy with all around :)